File: c++/programatm.cpp

#include <iostream>
#include <string>
#include <iomanip>
using namespace std;

bool validasiNama(string nama) {
    for (char c : nama) {
        if (!isalpha(c) && c != ' ') {
            return false;
        }
    }
    return !nama.empty();
}

int main() {
    string nama;
    double saldo;
 string nim;
    bool namaValid = false;
    
    while (!namaValid) {
        cout << "Masukkan nama: ";
        getline(cin, nama);
        
        namaValid = validasiNama(nama);
        
        if (!namaValid) {
            cout << "Nama tidak valid! Hanya boleh mengandung huruf dan spasi." << endl;
        }
    }
    cout << "Masukkan NIM (hanya angka): ";
    cin >> nim;
    
    saldo = stod(nim);
    
    cout << "\n=== DATA AWAL ===" << endl;
    cout << "Nama: " << nama << endl;
    cout << "Saldo Awal: " << (saldo) << endl;
    
    int pilihan;
    double jumlah;
    string rekeningTujuan;
    
    do {
cout << "\n=== MENU ATM ===" << endl;
        cout << "1. Cek Saldo" << endl;
        cout << "2. Tarik Tunai" << endl;
        cout << "3. Setor Tunai" << endl;
        cout << "4. Transfer" << endl;
        cout << "5. Keluar" << endl;
        cout << "Pilih menu (1-5): ";
        cin >> pilihan;
        
        switch (pilihan) {
            case 1:
                cout << "\n--- CEK SALDO ---" << endl;
                cout << "Saldo Anda: " << (saldo) << endl;
                break;
                
            case 2:
                cout << "\n--- TARIK TUNAI ---" << endl;
                cout << "Saldo tersedia: " << (saldo) << endl;
                cout << "Masukkan jumlah penarikan: ";
                cin >> jumlah;
                
   if (jumlah <= 0) {
                    cout << "Jumlah tidak valid!" << endl;
                } else if (jumlah > saldo) {
                    cout << "Saldo tidak mencukupi!" << endl;
                } else {
                    saldo -= jumlah;
                    cout << "Penarikan berhasil!" << endl;
                    cout << "Jumlah ditarik: " << (jumlah) << endl;
                    cout << "Saldo sisa: " << (saldo) << endl;
                }
                break;
                
            case 3:
                cout << "\n--- SETOR TUNAI ---" << endl;
                cout << "Masukkan jumlah setoran: ";
                cin >> jumlah;
                
                if (jumlah <= 0) {
                    cout << "Jumlah tidak valid!" << endl;
                } else {
                    saldo += jumlah;
                    cout << "Setoran berhasil!" << endl;
                    cout << "Jumlah disetor: " << (jumlah) << endl;
                    cout << "Saldo sekarang: " << (saldo) << endl;
                }
                break;
                
            case 4:
 cout << "\n--- TRANSFER ---" << endl;
                cout << "Saldo Anda: " << (saldo) << endl;
                cout << "Masukkan nomor rekening tujuan: ";
                cin >> rekeningTujuan;
                cout << "Masukkan jumlah transfer: ";
                cin >> jumlah;
                
                if (jumlah <= 0) {
                    cout << "Jumlah tidak valid!" << endl;
                } else if (jumlah > saldo) {
                    cout << "Saldo tidak mencukupi!" << endl;
                } else {
                    saldo -= jumlah;
                    cout << "Transfer berhasil!" << endl;
                    cout << "Ke rekening: " << rekeningTujuan << endl;
                    cout << "Jumlah: " << (jumlah) << endl;
                    cout << "Saldo sisa: " << (saldo) << endl;
                }
                break;
                
            case 5:
                cout << "\nTerima kasih " << nama << " telah menggunakan ATM!" << endl;
                break;
                
            default:
                cout << "Pilihan tidak valid! Silakan pilih 1-5." << endl;
                break;
        }
        
    } while (pilihan != 5);

    return 0;
}

File: Java/programatm.java
import java.util.Scanner;

public class programatm {
    static Scanner scanner = new Scanner(System.in);

    // Validasi nama: hanya huruf dan spasi
    static boolean validasiNama(String nama) {
        return nama.matches("^[a-zA-Z\\s]+$");
    }

    public static void main(String[] args) {
        String nama;
        boolean statusNama = false;
        String nim;
        long saldo = 0;

        // 1 & 2 & 3 — Input dan validasi nama
        do {
            System.out.print("Masukkan Nama: ");
            nama = scanner.nextLine();

            statusNama = validasiNama(nama);
            System.out.println("Status nama: " + statusNama); // TRUE / FALSE

            if (!statusNama) {
                System.out.println("Nama tidak valid! Masukkan lagi.\n");
            }

        } while (!statusNama);

        // 4 & 5 — Input NIM dan konversi ke saldo
        System.out.print("Masukkan NIM (angka saja): ");
        nim = scanner.nextLine();

        try {
            saldo = Long.parseLong(nim); 
        } catch (NumberFormatException e) {
            System.out.println("NIM tidak valid! Saldo diset default 1000000");
            saldo = 1000000;
        }

        // 6 — Tampilkan data awal
        System.out.println("\n=== DATA AWAL ===");
        System.out.println("Nama: " + nama);
        System.out.println("Saldo awal: " + saldo);

        // 7 — Menu ATM (switch-case)
        int pilihan;

        do {
            System.out.println("\n=== MENU ATM ===");
            System.out.println("1. Cek Saldo");
            System.out.println("2. Tarik Tunai");
            System.out.println("3. Setor Tunai");
            System.out.println("4. Keluar");
            System.out.print("Pilih menu: ");

            try {
                pilihan = Integer.parseInt(scanner.nextLine());
            } catch (Exception e) {
                pilihan = 0;
            }

            switch (pilihan) {
                case 1:
                    System.out.println("\nSaldo Anda sekarang: " + saldo);
                    break;

                case 2:
                    System.out.print("\nMasukkan jumlah penarikan: ");
                    long tarik = Long.parseLong(scanner.nextLine());
                    if (tarik > saldo) {
                        System.out.println("Saldo tidak mencukupi!");
                    } else if (tarik <= 0) {
                        System.out.println("Jumlah tidak valid!");
                    } else {
                        saldo -= tarik;
                        System.out.println("Penarikan berhasil! Saldo sisa: " + saldo);
                    }
                    break;

                case 3:
                    System.out.print("\nMasukkan jumlah setoran: ");
                    long setor = Long.parseLong(scanner.nextLine());
                    if (setor <= 0) {
                        System.out.println("Jumlah tidak valid!");
                    } else {
                        saldo += setor;
                        System.out.println("Setoran berhasil! Saldo baru: " + saldo);
                    }
                    break;

                case 4:
                    System.out.println("\nTerima kasih telah menggunakan ATM!");
                    break;

                default:
                    System.out.println("Pilihan tidak valid!");
            }

        } while (pilihan != 4);
    }
}